index.js

```diff
@@ -597,11 +597,13 @@ async function apostrophe(options, telemetry, rootSpan) {
 
     // The base module sort metadata
     for (const name of moduleNames) {
-      if (definitions[name].before) {
+      const metadata = await self.synth.getMetadata(name);
+      const before = Object.values(metadata.before).find(name => name !== undefined);
```

Comment:
So basically, if I extend a module that has `before`, that module will also go to that `before`. Which I think makes sense. A chain reorder.